receiver/sqlserverreceiver/scraper.go

```diff
@@ -831,11 +831,25 @@ func vanillaRetriever(row sqlquery.StringMap, columnName string) (any, error) {
 
 func retrieveInt(row sqlquery.StringMap, columnName string) (any, error) {
 	var err error
-	result := 0
+	var result int64
 	if row[columnName] != "" {
-		result, err = strconv.Atoi(row[columnName])
+		result, err = strconv.ParseInt(row[columnName], 10, 64)
+		if err != nil {
+			// SQL Server stores large integers in scientific e notation
+			// (eg 123456 is stored as 1.23456e+5)
+			// This value cannot be parsed by strconv.ParseInt, but is successfully
+			// parsed by strconv.ParseFloat. The goal is here to convert to int
+			// even if the stored value is in scientific e notation.
+			var resultFloat float64
+			resultFloat, err = strconv.ParseFloat(row[columnName], 64)
+			if err == nil {
+				result = int64(resultFloat)
+			}
+		}
+	} else {
+		err = fmt.Errorf("no value found for column %s", columnName)
```

Comment:
Technically in both `retrieveInt` and `retrieveFloat` I've added new potential errors scenarios, but both of these methods are only called currently from a select when a column name is seen in a row. I don't believe either of the new error cases are hit today, they're an attempt at future-proofing these methods.

Also, even if they're introducing new errors, the receiver is already failing to scrape the given metric or attribute, but it's just being hidden here and metrics are recorded as zero values. The proper behavior should be to notify users of errors.